api/v1alpha1/clusterconfig_types.go

```diff
@@ -36,20 +35,53 @@ const (
 
 // ClusterConfigSpec defines the desired state of ClusterConfig
 type ClusterConfigSpec struct {
-	cro.ClusterRelocationSpec `json:",inline"`
+	ClusterInfo `json:",inline"`
 
-	// BareMetalHostRef identifies a BareMetalHost object to be used to attach the configuration to the host.
-	// +optional
-	BareMetalHostRef *BareMetalHostReference `json:"bareMetalHostRef,omitempty"`
+	// PullSecretRef is a reference to new cluster-wide pull secret.
+	// If defined, it will replace the secret located at openshift-config/pull-secret.
+	// The type of the secret must be kubernetes.io/dockerconfigjson.
+	PullSecretRef *corev1.LocalObjectReference `json:"pullSecretRef,omitempty"`
+
+	// CABundle is a reference to a config map containing the new bundle of trusted certificates for the host.
+	// The tls-ca-bundle.pem entry in the config map will be written to /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
+	CABundleRef *corev1.LocalObjectReference `json:"caBundleRef,omitempty"`
 
 	// NetworkConfigRef is the reference to a config map containing network configuration files if necessary.
 	// Keys should be of the form *.nmconnection and each represent an nmconnection file to be applied to the host.
 	// +optional
 	NetworkConfigRef *corev1.LocalObjectReference `json:"networkConfigRef,omitempty"`
 
-	// ExtraManifestsRef is a reference to a config map containing additional manifests to be applied to the relocated cluster.
+	// ExtraManifestsRefs is list of config map references containing additional manifests to be applied to the relocated cluster.
 	// +optional
-	ExtraManifestsRef *corev1.LocalObjectReference `json:"extraManifestsRef,omitempty"`
+	ExtraManifestsRefs []corev1.LocalObjectReference `json:"extraManifestsRef,omitempty"`
+
+	// BareMetalHostRef identifies a BareMetalHost object to be used to attach the configuration to the host.
+	// +optional
+	BareMetalHostRef *BareMetalHostReference `json:"bareMetalHostRef,omitempty"`
+}
+
+// ClusterInfo struct that describe current cluster critical info
+type ClusterInfo struct {
```

Comment:
Maybe let me make it pkg in LCA?

Reply:
Yeah, I was debating an import, but didn't attempt it.

LCA doesn't actually seem that hard to import, is this exported there?

Reply:
Looks public. I can probably just try to import.

ref: https://github.com/openshift-kni/lifecycle-agent/blob/37291358c89d72ecf7b8ae1e37f752d901a729b9/ibu-imager/clusterinfo/clusterinfo.go#L30

Reply:
We should. If needed i will move it where you will need it :)

Reply:
Done